packages/ui/src/common/components/Search/SearchResultsModal.tsx

```diff
@@ -19,16 +19,23 @@ import { ModalWithDataCall } from '@/common/providers/modal/types'
 import { ThreadItemBreadcrumbs } from '@/forum/components/threads/ThreadItemBreadcrumbs'
 import { ForumRoutes } from '@/forum/constant'
 
+import { useDebounce } from '../../hooks/useDebounce'
+
 export type SearchResultsModalCall = ModalWithDataCall<'SearchResults', { search: string }>
 
 export const SearchResultsModal = () => {
   const { hideModal, modalData } = useModal<SearchResultsModalCall>()
-
   const [search, setSearch] = useState(modalData.search)
   const [activeTab, setActiveTab] = useState<SearchKind>('FORUM')
   const { forum, forumPostCount, isLoading } = useSearch(search, activeTab)
-  const pattern = useMemo(() => (search ? RegExp(escapeStringRegexp(search), 'ig') : null), [search])
-
+  const isValid = () => !debouncedSearch || debouncedSearch.length === 0 || debouncedSearch.length > 2
+  const debouncedSearch = useDebounce(search, 400)
+  const [pattern, setPattern] = useState<RegExp | null>(null)
+  useEffect(() => {
+    if (isValid()) {
```

Comment:
@oleksanderkorn sorry I had missed that if `debouncedSearch` is empty `isValid()` is `true`:
```suggestion
    if (debouncedSearch && isValid()) {
```